pyproject.toml

```diff
@@ -149,6 +149,7 @@ pre-commit = "^3.7.0"
 ipykernel = "^6.29.4"
 semver = "^3.0.2"
 pillow = "^10.1.0"
+litellm = {version = "1.51.0", extras = ["proxy"]}
```

Comment:
The `[proxy]` extra is required in order to run the LiteLLM server. Adding it as a dev dependency here. This resulted in a fairly hefty lockfile update. @okhat let me know if there are concerns here - we can always add it elsewhere (e.g. requirements-dev.txt)